mutiny-core/src/onchain.rs

```diff
@@ -115,19 +120,29 @@ impl<S: MutinyStorage> OnChainWallet<S> {
             .await?;
 
         // get new wallet lock for writing and apply the update
-        match self.wallet.try_write() {
-            Ok(mut wallet) => match wallet.apply_update(update) {
-                Ok(_) => wallet.commit()?,
+        loop {
+            match self.wallet.try_write() {
+                Ok(mut wallet) => match wallet.apply_update(update.clone()) {
```

Comment:
Instead of cloning update, you could make the `Ok(_) => wallet.commit()?` condition:

```
                    Ok(_) => {
                        wallet.commit()?;
                        return Ok(());
                    }
```

Reply:
good idea